src/lightning/pytorch/trainer/trainer.py

```diff
@@ -1023,6 +1023,9 @@ def _run_stage(self) -> Optional[Union[_PREDICT_OUTPUT, _EVALUATE_OUTPUT]]:
         # wait for all to join if on distributed
         self.strategy.barrier("run-stage")
 
+        # TODO: Remove `set_to_none=True` when PyTorch 2.0 is the minimum supported version
+        self.lightning_module.zero_grad(set_to_none=True)
```

Comment:
nit: one easy way to avoid having to remember about this TODO is to condition the call based on `_TORCH_GREATER_EQUAL_2_0`,